src/test/java/io/openliberty/tools/common/config/ServerConfigDocumentTest.java

```diff
@@ -0,0 +1,218 @@
+    @Test
+    public void processServerXml() throws FileNotFoundException, IOException, XPathExpressionException, SAXException {
+        File serversDir = new File(RESOURCES_DIR + DEFAULT_SERVER_DIR);
+        Document doc;
+
+        // no variables defined
+        ServerConfigDocument configDocument = new ServerConfigDocument(new TestLogger());
+        File empty = new File(serversDir, "emptyList.xml");
+        doc = configDocument.parseDocument(empty);
+        configDocument.parseVariablesForBothValues(doc);
+        assertTrue(configDocument.getDefaultProperties().isEmpty() && configDocument.getProperties().isEmpty());
+
+        // variables defined
+        configDocument = new ServerConfigDocument(new TestLogger());
+        File defined = new File(serversDir, "definedVariables.xml");
+        doc = configDocument.parseDocument(defined);
+        configDocument.parseVariablesForBothValues(doc);
+        assertEquals(1, configDocument.getDefaultProperties().size());
+        assertEquals("9080", configDocument.getDefaultProperties().getProperty("default.http.port"));
+        assertEquals(1, configDocument.getProperties().size());
+        assertEquals("9081", configDocument.getProperties().getProperty("http.port"));
+
+        // variables defined in <include/> files
+        configDocument = new ServerConfigDocument(new TestLogger());
+        configDocument.initializeFields(new TestLogger(), new File(serversDir, "server.xml"), null, null);
+        File include = new File(serversDir, "testIncludeParseVariables.xml");
+        doc = configDocument.parseDocument(include);
+        assertTrue(configDocument.getDefaultProperties().isEmpty() && configDocument.getProperties().isEmpty());
+        configDocument.parseIncludeVariables(doc);
+        assertEquals(1, configDocument.getDefaultProperties().size());
+        assertEquals("9080", configDocument.getDefaultProperties().getProperty("default.http.port"));
+        assertEquals(1, configDocument.getProperties().size());
+        assertEquals("9081", configDocument.getProperties().getProperty("http.port"));
+
+        // server.xmls read in increasing precedence
+        // TODO: refactor
+        // configDropins/defaults
+        // server.xml
+        // configDropins/overrides
+    }
+
+    // server.env files are read in increasing precedence
+    //   1. {wlp.install.dir}/etc
+    //   2. {wlp.user.dir}/shared
+    //   3. {server.config.dir}
+    // Table of directories: https://openliberty.io/docs/latest/reference/directory-locations-properties.html
+    @Test
+    public void processServerEnv() throws FileNotFoundException, Exception {
+        File serverDir = new File(RESOURCES_DIR + WLP_USER_DIR + DEFAULT_USER_SERVER_DIR);
+        File specificFile = new File(serverDir, "server2.env");
+        ServerConfigDocument configDocument = new ServerConfigDocument(new TestLogger());
+        configDocument.initializeFields(new TestLogger(), null, serverDir, null);
```

Comment:
I think you should consider passing in the Map<String,File> for the last parameter on `initializeFields` with correct values set for all the pre-defined Liberty variables. Then the `processServerEnv` would be able to do the right thing for precedence related to the number 1 and 2 locations.